docs/DeveloperGuide.adoc

```diff
@@ -153,94 +181,369 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+types (i.e Add, Delete, Edit etc), the corresponding variant parsers will be invoked (i.e `AddRecipeCommandParser`,
+`DeleteRecipeCommandParser` etc). After which, the corresponding command will be executed (i.e `AddRecipeCommand`,
+`DeleteRecipeCommand` etc).
+
+The figure below describes the execution of an `DeleteRecipeCommand`.
+The input provided is `delete recipe 1`.
+
+.DeleteRecipeCommand Sequence Diagram
+image::DeleteSequenceDiagram.png[]
+
+After a successful execution, the recipe with the specified index will be deleted from the recipe book.
+
+/ maybe insert a screenshot to show the expected output.
+
+The current implementation makes use of an Sets to store the `Ingredient` objects.
+This is the easiest way to implement the `Ingredient` component of the recipe as it builds on the existing codebase
+of ab3.
+
+==== Alternatives
+An alternative would be to use an ArrayList instead of Sets to store `Ingredient` objects. This would make the
+implementation of `Ingredient` specific objects easier as they could be referenced through indexing of the ArrayList,
+which is more straight forward.
 
-Step 5. The user then decides to execute the command `list`. Commands that do not modify the address book, such as `list`, will usually not call `Model#commitAddressBook()`, `Model#undoAddressBook()` or `Model#redoAddressBook()`. Thus, the `addressBookStateList` remains unchanged.
+==== Our Choice
+We will soon implement ArrayLists instead as the current implementation of `Ingredient` objects using Sets do not allow
+for ease of use for the user whenever the user wishes to edit `Ingredient` objects present within a `Recipe`.
 
-image::UndoRedoState4.png[]
+=== Workout Planner feature
+==== Implementation
+===== Exercise Management
 
-Step 6. The user executes `clear`, which calls `Model#commitAddressBook()`. Since the `currentStatePointer` is not pointing at the end of the `addressBookStateList`, all address book states after the `currentStatePointer` will be purged. We designed it this way because it no longer makes sense to redo the `add n/David ...` command. This is the behavior that most modern desktop applications follow.
+Every workout comprises of varying number of exercises. Users can use existing exercises or create their own custom
+exercises. Every exercise contains the following information:
 
-image::UndoRedoState5.png[]
+* `ExerciseName` representing the name of the exercise
+* `MusclesTrained` comprising of the primary `MuscleType` as well as an ArrayList of secondary `MuscleType` trained
+* `Intensity` or how demanding the exercise is
+* A set of `ExerciseDetails` which are optional additional information of the exercise such as `ExerciseWeight`,
+`Distance`, `Sets` and `Repetitions`.
 
-The following activity diagram summarizes what happens when a user executes a new command:
+The Exercise class is represented by the following class diagram below.
+
+image::ExerciseClassDiagram.png[]
```

Comment:
Should choose to show association using attribute or with arrow, but not both.
There are also too many details in this diagram. Private attributions don't have to be shown. 🙈